### ISSUE TITLE
calc_sogs_after_sos_restored has a bug.

### ISSUE THREAD

**Issue by kanatatsu64** [othero/othero#35]
# Problem

There is a bug when calc_sogs_after_sos_restored is called to restore the sos to SOS.BLANK. This is because the fact is overlooked that restoration of a certain position only causes soss around the position to flip between SOS.DARK and SOS.LIGHT. The restoration does not restore SOS.DARK or SOS.LIGHT to SOS.BLANK vice versa.

The sos restoration is made here:

https://github.com/othero/othero/blob/dbf3c9386ad981315c3c417843868ccb7a2e65c0/othero/rule/libbackward.py#L66

The implementation is too simple to deal with the fact explained above.

# Example

Add test_calc_sogs_after_sos_restored1 test shown below to TestRuleBackward class. The test will fail.

First of all, I quote TestRuleBackward.__SOG from test/rule/test_libbackward.py for reference purpose.

```python
__SOG = [ \
    [libtypes.SOS.BLANK, libtypes.SOS.DARK , libtypes.SOS.LIGHT, libtypes.SOS.DARK ,], \
    [libtypes.SOS.DARK , libtypes.SOS.LIGHT, libtypes.SOS.BLANK, libtypes.SOS.BLANK,], \
    [libtypes.SOS.DARK , libtypes.SOS.LIGHT, libtypes.SOS.BLANK, libtypes.SOS.BLANK,], \
    [libtypes.SOS.DARK , libtypes.SOS.DARK , libtypes.SOS.DARK , libtypes.SOS.BLANK] \
]
```

```python
def test_calc_sogs_after_sos_restored1(self):
    def is_equal(x, y):
        return all([a in y for a in x]) \
                and all([a in x for a in y])

    result = is_equal(
        libbackward.calc_sogs_after_sos_restored(self.__SOG, (3, 0), libtypes.SOS.BLANK),
        [
            [
                [libtypes.SOS.BLANK, libtypes.SOS.DARK , libtypes.SOS.LIGHT, libtypes.SOS.DARK ],
                [libtypes.SOS.DARK , libtypes.SOS.LIGHT, libtypes.SOS.BLANK, libtypes.SOS.BLANK],
                [libtypes.SOS.LIGHT, libtypes.SOS.LIGHT, libtypes.SOS.BLANK, libtypes.SOS.BLANK],
                [libtypes.SOS.BLANK, libtypes.SOS.DARK , libtypes.SOS.DARK , libtypes.SOS.BLANK]
            ],
            [
                [libtypes.SOS.BLANK, libtypes.SOS.DARK , libtypes.SOS.LIGHT, libtypes.SOS.DARK ],
                [libtypes.SOS.DARK , libtypes.SOS.LIGHT, libtypes.SOS.BLANK, libtypes.SOS.BLANK],
                [libtypes.SOS.DARK , libtypes.SOS.LIGHT, libtypes.SOS.BLANK, libtypes.SOS.BLANK],
                [libtypes.SOS.BLANK, libtypes.SOS.LIGHT, libtypes.SOS.DARK , libtypes.SOS.BLANK]
            ],
            [
                [libtypes.SOS.BLANK, libtypes.SOS.DARK , libtypes.SOS.LIGHT, libtypes.SOS.DARK ],
                [libtypes.SOS.DARK , libtypes.SOS.LIGHT, libtypes.SOS.BLANK, libtypes.SOS.BLANK],
                [libtypes.SOS.LIGHT, libtypes.SOS.LIGHT, libtypes.SOS.BLANK, libtypes.SOS.BLANK],
                [libtypes.SOS.BLANK, libtypes.SOS.LIGHT, libtypes.SOS.DARK , libtypes.SOS.BLANK]
            ]
        ]
    )
    self.assertTrue(result)
```

**Comment by kanatatsu64** [comment]
After fixing the bug, add the test case described in the Example section.

**Comment by kanatatsu64** [comment]
Fixed in 9bca3c3.
